crates/bundler/src/bundler.rs

```diff
@@ -1,40 +1,151 @@
 use ethers::{
-    prelude::SignerMiddleware,
+    prelude::{LocalWallet, SignerMiddleware},
     providers::{Http, Middleware, Provider},
     signers::Signer,
-    types::{transaction::eip2718::TypedTransaction, Address, H256},
+    types::{
+        transaction::eip2718::TypedTransaction, Address, Eip1559TransactionRequest, H256, U64,
+    },
+};
+use ethers_flashbots::{
+    BundleRequest, FlashbotsMiddleware, PendingBundleError::BundleNotIncluded, SimulatedBundle,
 };
 use silius_contracts::entry_point::EntryPointAPI;
-use silius_primitives::{Chain, UserOperation, Wallet};
+use silius_primitives::{consts::flashbots_relay_endpoints, Chain, UserOperation, Wallet};
 use std::{sync::Arc, time::Duration};
 use tracing::{info, trace};
+use url::Url;
+
+/// A type alias for the Flashbots Signer client
+type FlashbotsClientType<M> =
+    Arc<SignerMiddleware<FlashbotsMiddleware<M, LocalWallet>, LocalWallet>>;
 
-#[derive(Clone)]
+/// A type alias for the Ethereum Signer client
+type EthClientType<M> = Arc<SignerMiddleware<M, LocalWallet>>;
+
+/// The `SendBundleMode` determines whether to send the bundle to a Ethereum execution client or to Flashbots relay
+#[derive(Clone, Debug, PartialEq)]
+pub enum SendBundleMode {
+    /// Send the bundle to a Ethereum execution client
+    EthClient,
```

Comment:
Can we add to the enum Vec<String> to each variant>
EthClient(Vec<String>)
Flashbots(Vec<String>)

First contain list of eth execution client nodes, and the second flashbots relay endpoints. That way we can pass the string with enum and have less properties in struct Bundler

Reply:
I see the point of using multiple relay endpoints, but I don't see why we need multiple execution client nodes. Each bundler should only use one execution client node. 

Is there a scenario where a bundler would use multiple execution client nodes?

Reply:
Ok, you're right. Doesn't make a lot of sense. Let's leave it like that :+1: